plenum/test/txn_author_agreement/conftest.py

```diff
@@ -33,25 +36,49 @@
 
 
 # Note. tconf is necessary for proper config initialization
-@pytest.fixture
+@pytest.fixture(scope="module")
 def config_ledger(tconf, tmpdir_factory):
     tdir = tmpdir_factory.mktemp('').strpath
     return Ledger(CompactMerkleTree(),
                   dataDir=tdir)
 
 
-@pytest.fixture
+@pytest.fixture(scope="module")
 def config_state():
     return PruningState(KeyValueStorageInMemory())
 
 
-@pytest.fixture
-def config_req_handler(config_state,
-                       config_ledger):
-    return ConfigReqHandler(config_ledger,
-                            config_state,
-                            domain_state=FakeSomething(),
-                            bls_store=FakeSomething())
+@pytest.fixture(scope="module")
+def db_manager(config_state, config_ledger):
+    db = DatabaseManager()
+    db.register_new_database(CONFIG_LEDGER_ID, config_ledger, config_state)
+    return db
+
+
+@pytest.fixture(scope="module")
+def write_manager(db_manager):
+    wm = WriteRequestManager(db_manager)
+    taa_handler = TxnAuthorAgreementHandler(db_manager)
+    taa_aml_handler = TxnAuthorAgreementAmlHandler(db_manager)
+    wm.register_req_handler(taa_handler)
+    wm.register_req_handler(taa_aml_handler)
+    return wm
+
+
+@pytest.fixture(scope="module")
+def taa_handler(write_manager):
+    for h_l in write_manager.request_handlers.values():
+        for h in h_l:
+            if isinstance(h, TxnAuthorAgreementHandler):
+                return h
```

Comment:
Having a generic getter in write_manager (even if it is just for tests) would be more nice

Reply:
It will be replaced with helper method.